edx_proctoring/api.py

```diff
@@ -877,6 +877,44 @@ def is_state_transition_legal(from_status, to_status):
     return True
 
 
+def can_update_credit_grades_and_email(attempts, to_status):
+    """
+    Determine and return as a boolean whether an attempt should trigger an update to credit and grades
+
+    Arguments:
+        attempts: a list of all currently active attempts for a given user_id and exam_id
+        to_status: future status of a proctored exam attempt
+    """
+    statuses = [attempt['status'] for attempt in attempts]
+    if len(statuses) == 1:
+        # if there is only one attempt for a user in an exam, it can be responsible for updates to credits and grades
+        return True
+    if to_status in ['declined', 'error']:
```

Comment:
I think you should refer to the `ProctoredExamStudentAttemptStatus` class directly here and elsewhere in this function.

Reply:
Noted, will udpate